auth/src/oauth2.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{Error, Result};
+use chrono::Utc;
+use serde::Serialize;
+
+/// JSON Web Signature for a token.
+#[derive(Serialize)]
+pub(crate) struct JwsClaims<'a> {
+    pub(crate) iss: &'a str,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) scope: Option<&'a str>,
+    pub(crate) aud: &'a str,
+    pub(crate) exp: Option<i64>,
+    pub(crate) iat: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) typ: Option<&'a str>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) sub: Option<&'a str>,
+}
+
+impl JwsClaims<'_> {
+    pub(crate) fn encode(&mut self) -> Result<String> {
```

Comment:
I'm very surprised that the `encode()` method mutates the struct.

Reply:
It does so to provide good defaults as this struct will be directly encoded into a string. It could probably just consume self if you think that is better.